// mongodb/productDao.js
// dao/mongo/productDao.js
const Product = require('./models/product');

class ProductManager {
    async addProduct(product) {
        try {
            const newProduct = new Product(product);
            await newProduct.save();
            console.log(`Produto ${product.title} adicionado com sucesso`);
        } catch (error) {
            console.error('Erro ao adicionar produto:', error.message);
        }
    }

    async getProducts() {
        try {
            return await Product.find();
        } catch (error) {
            console.error('Erro ao obter produtos:', error.message);
            return [];
        }
    }

    async getProductById(id) {
        try {
            return await Product.findById(id);
        } catch (error) {
            console.error('Erro ao obter produto:', error.message);
            return null;
        }
    }

    async updateProduct(id, updatedProduct) {
        try {
            await Product.findByIdAndUpdate(id, updatedProduct, { new: true });
            console.log(`Produto com ID ${id} atualizado com sucesso`);
        } catch (error) {
            console.error('Erro ao atualizar produto:', error.message);
        }
    }

    async deleteProduct(id) {
        try {
            await Product.findByIdAndDelete(id);
            console.log(`Produto com ID ${id} deletado com sucesso`);
        } catch (error) {
            console.error('Erro ao deletar produto:', error.message);
        }
    }
}

module.exports = new ProductManager();
